### ISSUE TITLE
Product List: sometimes duplicate products appear after applying filters and resuming the app from background

### ISSUE THREAD

**Issue by designsimply** [woocommerce/woocommerce-android#3529]
Steps to reproduce:

1. Go to Products.
1. Filter for Z to A.
1. Background the app.
1. Open the app.
1. Filter for A to Z. 
1. Expect each product to only show up once in the list.

**Result:** products sometimes appear duplicated many times in the product list when resuming the app from the background with filters applied previously. ([1m11s](https://cloudup.com/cb0vNhJGXZp))

<img src="https://user-images.githubusercontent.com/1119271/106949707-8bdf9c00-66ea-11eb-89e2-8afee031b5d4.png" width="270" alt="Screen Shot 2021-02-04 at Feb 4 1 11 36 PM">
<sup>Tested with WCAndroid 5.9-rc-3 on Pixel 3 Android 11 on comparative-moth.jurassic.ninja (WP 5.6 JP 9.4 WC 5.0.0-rc.1) logged in as testuser01121.</sup>

```
[Feb-04 20:00 UTILS i] 🔵 Tracked: main_tab_products_selected, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"ProductListFragment"}
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_view_sorting_options_tapped, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_sorting_option_selected, Properties: {"order":"name,ascending","blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":187962955,"is_wpcom_store":false}
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Feb-04 20:00 PRODUCTS d] can't load more products
[Feb-04 20:01 UTILS i] 🔵 Tracked: application_closed
[Feb-04 20:01 WP d] API Dispatching action: WCStatsAction-FETCH_REVENUE_STATS_AVAILABILITY
[Feb-04 20:01 WP d] API Dispatching action: WCStatsAction-FETCH_REVENUE_STATS
[Feb-04 20:01 WP d] API Dispatching action: WCStatsAction-FETCH_NEW_VISITOR_STATS
[Feb-04 20:01 WP d] API Dispatching action: WCOrderAction-FETCH_HAS_ORDERS
[Feb-04 20:01 UTILS i] 🔵 Tracked: application_opened
```

**Comment by designsimply** [comment]
I spotted duplicates in the My Store > This Month > Top performers list periodically in v5.8 as well. In this case, the duplicates went away after I closed and re-opened the app.

Duplicates in Top performers|About|Duplicate gone after closing+re-opening the app
---|---|---|
![Screenshot_20210204-171041](https://user-images.githubusercontent.com/1119271/106971790-d83bd380-670c-11eb-97f0-d456518e955a.png)|![Screenshot_20210204-171058](https://user-images.githubusercontent.com/1119271/106971799-da059700-670c-11eb-8d76-a9561c776a67.png)|![Screenshot_20210204-171225](https://user-images.githubusercontent.com/1119271/106971800-da9e2d80-670c-11eb-8501-7fb9478a12b7.png)

**Comment by anitaa1990** [comment]
Hey @designsimply! Thank you for testing and flagging this issue! We did have an [issue](https://github.com/woocommerce/woocommerce-android/issues/3460) in the app in v5.8 where duplicate products were displayed in the top performers list. This was fixed in #3514 and scheduled for the 6.0 release. The issue was duplicate caching at the db level and it was in the products table so I'm not sure if this fixes the duplicate products issue as you mentioned. I'm not able to reproduce that duplicate products issue on any of my test sites 😓 

Would you be able to test from the latest `develop` to confirm whether this issue is fixed?

**Comment by anitaa1990** [comment]
This is fixed in #3514 in v6.0 so closing this one!
